src/ol/format/WKB.js

```diff
@@ -0,0 +1,866 @@
+      geometries = geometry.getGeometriesArray();
+    } else {
+      geometries = [geometry];
+    }
+    return geometries.map((geometry) => new Feature({geometry}));
+  }
+
+  /**
+   * Read a single geometry from a source.
+   *
+   * @param {string|ArrayBuffer|ArrayBufferView} source Source.
+   * @param {import("./Feature.js").ReadOptions} [opt_options] Read options.
+   * @return {import("../geom/Geometry.js").default} Geometry.
+   */
+  readGeometry(source, opt_options) {
+    const view = getDataView(source);
+    if (!view) {
+      return null;
+    }
+
+    const reader = new WkbReader(view);
+    const geometry = reader.readGeometry();
+
+    this.viewCache_ = view; // cache for internal subsequent call of readProjection()
+    const options = this.getReadOptions(source, opt_options);
+    this.viewCache_ = null; // release
+
+    return transformGeometryWithOptions(geometry, false, options);
+  }
+
+  /**
+   * Read the projection from a source.
+   *
+   * @param {string|ArrayBuffer|ArrayBufferView} source Source.
+   * @return {import("../proj/Projection.js").default|undefined} Projection.
+   */
+  readProjection(source) {
+    const view = this.viewCache_ || getDataView(source);
+    if (!view) {
+      return undefined;
+    }
+
+    const reader = new WkbReader(view);
+    reader.readWkbHeader();
+
+    return (reader.SRID && getProjection('EPSG:' + reader.SRID)) || undefined;
+  }
+
+  /**
+   * Encode a feature in this format.
+   *
+   * @param {import("../Feature.js").default} feature Feature.
+   * @param {import("./Feature.js").WriteOptions} [opt_options] Write options.
+   * @return {string|ArrayBuffer} Result.
+   */
+  writeFeature(feature, opt_options) {
+    return this.writeGeometry(feature.getGeometry(), opt_options);
+  }
+
+  /**
+   * Encode an array of features in this format.
+   *
+   * @param {Array<import("../Feature.js").default>} features Features.
+   * @param {import("./Feature.js").WriteOptions} [opt_options] Write options.
+   * @return {string|ArrayBuffer} Result.
+   */
+  writeFeatures(features, opt_options) {
+    return this.writeGeometry(
+      new GeometryCollection(features.map((f) => f.getGeometry())),
+      opt_options
+    );
+  }
+
+  /**
+   * Write a single geometry in this format.
+   *
+   * @param {import("../geom/Geometry.js").default} geometry Geometry.
+   * @param {import("./Feature.js").WriteOptions} [opt_options] Write options.
+   * @return {string|ArrayBuffer} Result.
+   */
+  writeGeometry(geometry, opt_options) {
+    const options = this.adaptOptions(opt_options);
+
+    const writer = new WkbWriter({
+      layout: this.layout,
+      littleEndian: this.littleEndian,
+      ewkb: this.ewkb,
+
+      nodata: {
+        Z: this.nodataZ,
+        M: this.nodataM,
+      },
+    });
+
+    // extract SRID from `dataProjection`
+    let SRID = Number.isInteger(this.SRID) ? Number(this.SRID) : null;
```

Comment:
If we want IE support, you could check `typeof this.SRID === 'boolean'` instead.

Reply:
`Number.isInteger` is included the the polyfills (`ol/source/IIIF` uses it to check that numbers are integer values).